### ISSUE TITLE
How to handle messages between actor system where one of them uses Phobos tracing?

### ISSUE THREAD

**Issue by object** [petabridge/phobos-issues#78]
This is a follow-up to https://github.com/petabridge/phobos-issues/issues/77.

Once I followed the advices from @Arkatufus and @Aaronontheweb I was able to exchange messages between two actor systems, one running from an F# script. But trace envelopes present a bigger problem when a remoting client is not instrumented with Phobos.

Here's a sample code using message of string type for simplicity:

```
let rec messageHandler (msg: obj) =
    match msg with
    | :? Phobos.Tracing.SpanEnvelope as msg' -> printfn $"Received string in envelope: {msg'.Message}"
    | _ -> printfn $"Received message of {msg.GetType()}"
    ignored ()

let monitor = spawn system "queue-monitor" <| props (actorOf messageHandler)

```

Basically, a remoting client must reference Phobos.Tracing package even if it doesn't use trace instrumentation, just to be able to parse the message. I wonder whether there is a way for a Phobos-instrumented actor to be selective about trace instrumentation, so remoting client that are unaware of OTel will be able to communicate with OTel-enabled systems.

Related to that: how will Akka Streams work between Phobos and non-Phobos actor systems? (Using StreamRefs).

**Comment by Aaronontheweb** [comment]
> Related to that: how will Akka Streams work between Phobos and non-Phobos actor systems? (Using StreamRefs).

Right now we disable Phobos from being injected into the usual Akka.Streams messages locally, but I haven't tested how that works over the network with `StreamRef`s - so that's a good idea.

One thing we had designed in Phobos 0.8 was a means of detecting whether or not a message came from a node that didn't have Phobos installed on it and responding accordingly in order to avoid this very problem, but we didn't implement it since there wasn't a huge amount of demand - we also weren't sure how reliable it would be in the grand scheme of things (i.e. requires a "dead man's switch" to detect non-installation.) 

For the time being it's probably best to just have the Phobos package installed if you're planning on using Akka.Remote to talk to something that is Phobos-enabled - so at the very least you can deserialize the messages coming back over the wire.

If in the future we design "header bits" that can be set in the remoting payloads, that's probably the most elegant solution to this problem as it circumvents the entire issue of having to swap serializers for OTEL in the first place. That's something we could probably achieve for v1.6 since we're already looking at replacing the networking and serialization systems altogether.

**Comment by object** [comment]
That would be really elegant if it's done with headers. For the time being we will be using Phobos packages in remoting clients.

**Comment by Aaronontheweb** [comment]
We have been able to validate that Phobos does not presently inject anything into `StreamRefs` with Akka.Streams - so those will work free from interference.
